test/ardop/test_log.c

```diff
@@ -0,0 +1,239 @@
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <setjmp.h>
+#include <string.h>
+#include <cmocka.h>
+
+#include <limits.h>
+#include <stdio.h>
+#include <time.h>
+
+#include "setup.h"
+
+#include "common/log.h"
+#include "common/log_file.h"
+
+int ardop_log_level_filter(const int level);
+
+/* Mocks */
+
+FILE* __wrap_fopen(
+	const char* restrict pathname,
+	const char* restrict mode
+) {
+	check_expected(pathname);
+	return (FILE*)mock();
+};
+
+FILE* __wrap_freopen(
+	const char* restrict pathname,
+	const char* restrict mode,
+	FILE* restrict handle
+) {
+	check_expected(pathname);
+	return (FILE*)mock();
```

Comment:
When compiling for 32-bit Linux or Windows (mingw32-make), lines 26 and 35 produce:
warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]

Seems to work though.

Reply:
This is because cmocka uses the "largest integral type" to communicate numbers across the mocking framework. On most 32-bit platforms, the pointer type is smaller than the largest integral type (usually 64 bits).

* Some of my test-cases inject mock pointer values.
* Sometimes we also check their values.

There should not be any problems from this since we widen valid pointers to 64 bits, then narrow them back down.

Regardless, I have fixed them with a combination of cmocka macros and changing how the some return values are used. ASAN remains happy with this fix on Linux 64.